smarts/core/sensors.py

```diff
@@ -486,6 +486,11 @@ def _agent_vehicle_is_not_moving(cls, sim, agent_id):
 
         return False
 
+    @classmethod
+    def clean_up(cls):
+        cls._oncoming_traffic_edge.cache_clear()
+        cls._vehicle_off_route_info.cache_clear()
```

Comment:
While `@lru_cache` is nice it looks like it can be useful to clear it in some cases.